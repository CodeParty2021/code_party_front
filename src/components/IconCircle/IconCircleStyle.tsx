import styled, { css } from "styled-components";
import { iconCircle } from "styles/colors";

export type IconCircleStyleProps = {};

const defaultStyle = css`
  display: flex;
  justify-content: center;
  align-items: center;
  background: ${iconCircle.bg};
  width: 64px;
  height: 64px;
  border-radius: 78.75px;
`;

export const IconCircleStyle = styled.div<IconCircleStyleProps>`
  ${defaultStyle}
`;

IconCircleStyle.defaultProps = {};
