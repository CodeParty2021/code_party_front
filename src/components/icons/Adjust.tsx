import React, { ComponentProps } from "react";
import {
  IconPrototype,
  IconsDir,
} from "./components/IconPrototype/IconPrototype";

type Props = {
  size?: 16 | 32;
} & Omit<ComponentProps<typeof IconPrototype>, "filename" | "size">;

const Adjust: React.FC<Props> = ({ size = 16, ...props }) => {
  return (
    <IconPrototype
      filename={`${IconsDir()}/adjust_${size}.svg`}
      size={`${size}px`}
      {...props}
    />
  );
};

export default Adjust;
