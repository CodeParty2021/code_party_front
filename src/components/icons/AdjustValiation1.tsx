import React, { ComponentProps } from "react";
import {
  IconPrototype,
  IconsDir,
} from "./components/IconPrototype/IconPrototype";

type Props = {
  size?: 16 | 32;
} & Omit<ComponentProps<typeof IconPrototype>, "filename" | "size">;

const AdjustValiation1: React.FC<Props> = ({ size = 16, ...props }) => {
  return (
    <IconPrototype
      filename={`${IconsDir()}/adjust_valiation1_${size}.svg`}
      size={`${size}px`}
      {...props}
    />
  );
};

export default AdjustValiation1;
