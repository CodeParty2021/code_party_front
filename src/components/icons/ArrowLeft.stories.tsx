import React from "react";
import { ComponentMeta, ComponentStory } from "@storybook/react";
import { ArrowLeft } from "./index";

export default {
  title: "Icons/ArrowLeft",
  component: ArrowLeft,
} as ComponentMeta<typeof ArrowLeft>;

const Template: ComponentStory<typeof ArrowLeft> = (args) => (
  <ArrowLeft {...args} />
);

export const Default = Template.bind({});
Default.args = {
  display: "inline-block",
  wrapper: "svg",
  fill: "#252525",
};

export const Orange = Template.bind({});
Orange.args = {
  ...Default.args,
  fill: "#FF5500",
};
