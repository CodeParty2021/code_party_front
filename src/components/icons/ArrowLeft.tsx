import React, { ComponentProps } from "react";
import {
  IconPrototype,
  IconsDir,
} from "./components/IconPrototype/IconPrototype";

type Props = {
  size?: 16 | 24;
} & Omit<ComponentProps<typeof IconPrototype>, "filename" | "size">;

const ArrowLeft: React.FC<Props> = ({ size = 16, ...props }) => {
  return (
    <IconPrototype
      filename={`${IconsDir()}/arrow_left_${size}.svg`}
      size={`${size}px`}
      {...props}
    />
  );
};

export default ArrowLeft;
