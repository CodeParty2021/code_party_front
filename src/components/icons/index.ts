export { default as ArrowLeft } from "./ArrowLeft";
export { default as ArrowRight } from "./ArrowRight";
export { default as Banner } from "./Banner";
export { default as Check } from "./Check";
export { default as CheckCircle } from "./CheckCircle";
export { default as Checked } from "./Checked";
export { default as Link } from "./Link";
export { default as Lock } from "./Lock";
export { default as Plus } from "./Plus";
export { default as Setting } from "./Setting";
export { default as Sword } from "./Sword";
export { default as Times } from "./Times";
export { default as TriangleBottom } from "./TriangleBottom";
export { default as Unlock } from "./Unlock";
