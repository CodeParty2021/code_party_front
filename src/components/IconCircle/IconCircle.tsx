import React from "react";
import { iconCircle } from "styles/colors";
import { Icon32 } from "types/utils";
import { IconCircleStyle, IconCircleStyleProps } from "./IconCircleStyle";

type Props = IconCircleStyleProps & {
  Icon: React.ComponentType<Icon32>;
};

export const IconCircle: React.FC<Props> = ({ Icon, ...styleProps }) => {
  return (
    <IconCircleStyle {...styleProps}>
      <Icon size={32} fill={iconCircle.icon} />
    </IconCircleStyle>
  );
};
