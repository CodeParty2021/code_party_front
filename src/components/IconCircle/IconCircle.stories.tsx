import React from "react";
import { ComponentStory, ComponentMeta } from "@storybook/react";

import { IconCircle } from "./IconCircle";
import { Plus } from "components/icons";

export default {
  title: "components/IconCircle",
  component: IconCircle,
} as ComponentMeta<typeof IconCircle>;

const Template: ComponentStory<typeof IconCircle> = (args) => (
  <IconCircle {...args} />
);

export const Default = Template.bind({});
Default.args = {
  Icon: Plus,
};
