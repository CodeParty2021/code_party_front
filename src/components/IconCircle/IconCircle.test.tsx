import React from "react";
import { IconCircle } from "./IconCircle";
import "jest-styled-components";
import { render } from "enzyme";
import { Plus } from "components/icons";

describe("<IconCircle />", () => {
  it("snapshot test", () => {
    const wrapper = render(<IconCircle Icon={Plus} />);

    expect(wrapper).toMatchSnapshot();
  });
});
