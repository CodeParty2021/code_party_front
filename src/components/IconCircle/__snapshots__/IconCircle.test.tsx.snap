// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<IconCircle /> snapshot test 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: #5FC322;
  width: 64px;
  height: 64px;
  border-radius: 78.75px;
}

.c1 {
  display: inline-block;
  width: 32px;
  height: 32px;
  fill: #FFF;
}

.c1 > span,
.c1 > div,
.c1 > svg {
  display: block;
  width: 100%;
  height: 100%;
}

<div
  class="c0"
>
  <span
    class="c1"
    display="inline-block"
    fill="#FFF"
  />
</div>
`;
